client.go

```diff
@@ -188,6 +192,35 @@ func (c *clientImpl) Setup(ctx context.Context, username, password, org, bucket
 	return response.JSON201, nil
 }
 
+func (c *clientImpl) SetupWithToken(ctx context.Context, username, password, org, bucket string, retentionPeriodHours int, token string) (*domain.OnboardingResponse, error) {
+	if username == "" || password == "" {
+		return nil, errors.New("a username and a password is required for a setup")
+	}
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	params := &domain.PostSetupParams{}
+	retentionPeriodSeconds := int64(retentionPeriodHours * 3600)
+	retentionPeriodHrs := int(time.Duration(retentionPeriodSeconds) * time.Second)
+	body := &domain.PostSetupJSONRequestBody{
+		Bucket:                 bucket,
+		Org:                    org,
+		Password:               &password,
+		RetentionPeriodSeconds: &retentionPeriodSeconds,
+		RetentionPeriodHrs:     &retentionPeriodHrs,
+		Username:               username,
+		Token:                  &token,
+	}
+	response, err := c.apiClient.PostSetupWithResponse(ctx, params, *body)
+	if err != nil {
+		return nil, err
+	}
+	if response.JSONDefault != nil {
+		return nil, domain.ErrorToHTTPError(response.JSONDefault, response.StatusCode())
+	}
+	c.httpService.SetAuthorization("Token " + *response.JSON201.Auth.Token)
+	return response.JSON201, nil
```

Comment:
Original `Setup(ctx context.Context, username, password, org, bucket string, retentionPeriodHours int)` should call this with an empty token to avoid two almost the same implementations. When the token is empty, do not apply it.